### PR TITLE
MODFQMMGR-136: adding organization entity type

### PR DESCRIPTION
**Pull request by kjain110**
Screen Shot 2024-02-14 at 4 04 03 PM
Screen Shot 2024-02-14 at 4 04 25 PM
Screen Shot 2024-02-14 at 4 26 13 PM

### CODE REVIEW

File: src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql [hunk 1]
```diff
@@ -0,0 +1,284 @@
+INSERT INTO entity_type_definition (id, derived_table_name, definition)
+    VALUES ('837f262e-2073-4a00-8bcc-4e4ce6e669b3','drv_organization_details', '{
+             "id": "837f262e-2073-4a00-8bcc-4e4ce6e669b3",
+             "name":"drv_organization_details",
+             "private" : false,
+             "fromClause" : "src_organizations AS org LEFT JOIN LATERAL jsonb_array_elements(org.jsonb->''accounts'') AS accounts(account_data) ON true",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:6]: By chance, are there any indexes in the organizations table that we can use in filterValueGetters for these columns?

Comment (kjain110) [reply src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:6]: @mweaver-ebsco good point totally forgot about those. will check and add

File: src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+             "columns": [
+                {
+                    "name": "last_updated",
+                    "dataType": {
+                      "dataType": "dateType"
+                           },
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:12]: Tiny nit: The formatting on these `dataType` blocks all look a little off

File: src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                                                   "label": "Cash",
+                                                   "value": "Cash"
+                                              },
+                                              {
+                                                    "label": "Physical Check",
+                                                    "value": "Physical Check"
+                                              },
+                                              {
+                                                   "label": "EFT",
+                                                   "value": "EFT"
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:95]: Sanity check: Should this be EFT or ETF? I've never heard of EFT, but ETF wouldn't really make a lot of sense either, so I just wanna be sure :P

Comment (ncovercash) [reply src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:95]: Electronic Funds Transfer

File: src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                              "dataType": "numberType"
+                                   },
+                            "valueGetter": "org.jsonb->>''discountPercent''",
+                            "visibleByDefault": false
+                 },
+                 {
+                            "name": "claiming_interval",
+                            "dataType": {
+                              "dataType": "integerType"
+                                   },
```
Comment (ncovercash) [inline src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:120]: formatting looks off here, too...may be worthwhile to copy/paste the whole blob through https://jsonbeautifier.org/ or similar

**Review by ncovercash — APPROVED** [review]

File: src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql (hunk 3 above)
Comment (kjain110) [reply src/main/resources/db/changelog/changes/v1.1.0/sql/entity-type-definitions/insert-organization-definition.sql:95]: @mweaver-ebsco , I double checked with database its ETF only :P

File: translations/mod-fqm-manager/en.json [hunk 5]
```diff
@@ -245,5 +245,35 @@
   "entityType.drv_purchase_order_line_details.po_updated_date": "PO updated date",
   "entityType.drv_purchase_order_line_details.vendor_code": "Vendor code",
   "entityType.drv_purchase_order_line_details.vendor_id": "Vendor ID",
-  "entityType.drv_purchase_order_line_details.vendor_name": "Vendor name"
+  "entityType.drv_purchase_order_line_details.vendor_name": "Vendor name",
+
+  "entityType.src_organization_type": "Organization types",
+  "entityType.src_organization_type.id": "ID",
+  "entityType.src_organization_type.organization_types_name": "Type",
```
Comment (mweaver-ebsco) [inline translations/mod-fqm-manager/en.json:252]: Do we need to prefix these with "Organization "?

**Pull request closed without merging**